### PR TITLE
Add SCD40/41 support

### PR DESCRIPTION
**Pull request by Arnold-n**
## Description:

Adds Sensirion SCD40/SCD41 CO2 sensor support.
**Related discussion:** https://github.com/arendst/Tasmota/discussions/11329

## Checklist:
  - [X] The pull request is done against the latest development branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [ ] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works with Tasmota core ESP32 V.1.0.7.3
  - [X] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

Code change is tested and works on ESP8266 with Tasmota 9.5.0.8, not sure which core version.

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by s-hadinger** [comment]
Thanks for your PR. You may need to wait for a week or two before Theo can merge

**Comment by Arnold-n** [comment]
Thanks s-hadinger!
According to sergios100 (https://github.com/arendst/Tasmota/discussions/11329), it works on ESP32 too.

**Comment by Jason2866** [comment]
Imho enabling both SCD30 AND SCD40 as default is to much. For example the SPS30 is not enabled at all. For backwards compatibility of the firmware, please disable the SCD40 driver.

**Comment by Arnold-n** [comment]
What's the reason not to enable it by default, code size vs expected interest? And do you want it excluded for the ESP32 image too? As you likely know the SCD30 and SCD40 (or SPS30) are unrelated devices: the SCD30 is NDIR while the SCD40 is a miniature sensor based on the photoacoustic sensing principle.

As another (newby) question, currently the SCD40xxx read operations return -1 in case of failure (and a data value otherwise), while write operations return -1 in case of failure or 0 otherwise - should I change the latter also to "or a data value otherwise" ?

**Comment by Jason2866** [comment]
Simple reason not all sensors can be enabled. It is Theos decision what will enabled by default.
The decision is based on what sensors are most used.

**Comment by Arnold-n** [comment]
Sorry this "merge branch" is a mistake. I will try to undo it.

File: tasmota/tasmota_configurations.h [hunk 1]
```diff
@@ -107,6 +107,7 @@
 //#define USE_MGC3130                            // [I2cDriver27] Enable MGC3130 Electric Field Effect Sensor (I2C address 0x42) (+2k7 code, 0k3 mem)
 //#define USE_MAX44009                           // [I2cDriver28] Enable MAX44009 Ambient Light sensor (I2C addresses 0x4A and 0x4B) (+0k8 code)
 #define USE_SCD30                              // [I2cDriver29] Enable Sensiron SCd30 CO2 sensor (I2C address 0x61) (+3k3 code)
+#define USE_SCD40                              // [I2cDriver62] Enable Sensiron SCd40 CO2 sensor (I2C address 0x62) (+3k5 code)
```
Comment (arendst) [inline tasmota/tasmota_configurations.h:110]: Pls disable as the image will become too big to support easy installation

Comment (Arnold-n) [reply tasmota/tasmota_configurations.h]: Disabled in sensors, not in ESP32, let me know if you intended to have that changed too.

**Comment by arendst** [comment]
Thx.

**Pull request merged**